### PR TITLE
Show details book

### PR DESCRIPTION
**Pull request by pdkproitf**

### CODE REVIEW

File: app/models/book.rb [hunk 1]
```diff
@@ -17,4 +17,12 @@ class Book < ActiveRecord::Base
   validates :pages, presence: true,
                           numericality: { only_integer: true, greater_than_or_equal_to: 1 }
 
+ def self.book_list(page)
+  Book.paginate(:page => page, :per_page => 12)
```
Comment (yoyocoder) [inline app/models/book.rb]: `12` should be constant.

Comment (yoyocoder) [inline app/models/book.rb]: Please use the new hash style.

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -4,12 +4,13 @@ class BooksController < ApplicationController
...
   # GET /books.json
   def index
-    @books = Book.all
+    @books = Book.book_list(params[:page])
   end
 
   # GET /books/1
   # GET /books/1.json
   def show
+    @book_list = Book.books_of_category(@book.category_id,@book.id)
```
Comment (yoyocoder) [inline app/controllers/books_controller.rb]: add space after `,`

File: app/models/book.rb [hunk 3]
```diff
@@ -17,4 +17,12 @@ class Book < ActiveRecord::Base
...
+ end
+
+ def self.books_of_category(category_id,book_id)
+  Book.where(:category_id => category_id).where.not(:id => book_id)
```
Comment (yoyocoder) [inline app/models/book.rb]: Please use the new hash style.

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -4,12 +4,13 @@ class BooksController < ApplicationController
...
   end
```
Comment (yoyocoder) [inline app/controllers/books_controller.rb]: Interested, What does this action do? It is show the book details, isn't it? If yes, what did the functions `books_of_category` do?

File: app/views/books/_signle_book.html.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    </div>
+
+    <div  data-toggle="tooltip" data-placement="top" title = "<%= book.title %>">
+      <h2>
+        <a href=<%= book_path(book) %>><%= book.title[0..15]  %> ...</a>
+      </h2>
+    </div>
+
+    <div class="product-carousel-price">
+      <ins>$<%= book.cost - (book.cost*book.sale) %></ins> <del>$<%= book.cost %></del>
```
Comment (yoyocoder) [inline app/views/books/_signle_book.html.erb]: add space before and after `*`

File: app/views/books/show.html.erb [hunk 6]
```diff
@@ -1,54 +1,132 @@
...
+        <div class="row">
+            <div class="col-md-4">
+                <div class="single-sidebar">
+                  <h2 class="sidebar-title">RELATED PRODUCTS</h2>
+                  <% @book_list.each do |book| %>
+                    <div class="thubmnail-recent">
+                        <img src="<%= book.photo %>" class="recent-thumb" alt="">
+                            <h2><a href=<%= book_path(book) %>> <%= book.title %> </a></h2>
+                            <div class="product-sidebar-price">
+                              <ins>$<%= book.cost - (book.cost*book.sale) %></ins> <del>$<%= book.cost %></del>
```
Comment (yoyocoder) [inline app/views/books/show.html.erb]: Same as above?

File: app/views/books/show.html.erb [hunk 7]
```diff
@@ -1,54 +1,132 @@
...
+                          <img src="<%= @book.photo %>" alt="">
+                        </div>
+                    </div>
+                </div>
+
+                <div class="col-sm-6">
+                    <div class="product-inner">
+                        <h2 class="product-name"><%= @book.title %></h2>
+                        <div class="product-inner-price">
+                            <ins>$<%= @book.cost - (@book.cost*@book.sale) %></ins> <del>$<%= @book.cost %></del>
```
Comment (yoyocoder) [inline app/views/books/show.html.erb]: same as above

File: app/views/books/show.html.erb [hunk 8]
```diff
@@ -1,54 +1,132 @@
...
+              </div>
+          </div>
+
+
+                        <div class="related-products-wrapper">
+                            <h2 class="related-products-title">Details Book</h2>
+                            <table class="table table-bordered table-detail table-striped">
+                              <tbody>
+                                <tr>
+                                <td> Tác Giả </td>
```
Comment (yoyocoder) [inline app/views/books/show.html.erb]: Please use english 😄

Comment (yoyocoder) [inline app/views/books/show.html.erb]: References: https://www.amazon.com/books-used-books-textbooks/b?ie=UTF8&node=283155

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (pdkproitf) [inline app/controllers/books_controller.rb]: i want to show list book have the same category on details book page.
you can see in heroku

**Comment by pdkproitf** [comment]
fixed at show_book_details_fix branch

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (yoyocoder) [inline app/controllers/books_controller.rb]: ` show list book have the same category on details book page` => You should do it in action `Index` (with params) or categories_controller.rb. One more time, please follow RESTful.

Comment (yoyocoder) [inline app/controllers/books_controller.rb]: action `show` in `books_controller` just should show the details of book.

Comment (pdkproitf) [inline app/controllers/books_controller.rb]: ok i got it. but i have a function in this page. and i think it good. may be you should review it. on details book. And after that if you still think it the bad idea i will change

Comment (yoyocoder) [inline app/controllers/books_controller.rb]: ` i have a function in this page. and i think it good. may be you should review it. on details book.` I cannot catch your mean ??

File: app/models/book.rb (hunk 3 above)
Comment (yoyocoder) [inline app/models/book.rb]: this function should be move to Category model

File: app/controllers/books_controller.rb (hunk 4 above)
Comment (yoyocoder) [inline app/controllers/books_controller.rb]: Okay, I got it. But I think you can do like this:
```ruby
@books = @book.category.books
```

and in books_controller.rb/set_book
```ruby
@book = Book.find(params[:id]).includes(category: :books)
```

Comment (pdkproitf) [inline app/controllers/books_controller.rb]: @book = Book.find(params[:id]).includes(category: :books)
it not working. errr=> " undefines includes "

Comment (yoyocoder) [inline app/controllers/books_controller.rb]: How about `@book = Book.includes(category: :books).find(params[:id])`?

**Pull request closed without merging**